ocp_resources/machineset.py

```diff
@@ -0,0 +1,225 @@
+
+    def wait(self, timeout=TIMEOUT, sleep=1):
+        """
+        Wait for machine-set to reach 'ready' state.
+
+        Args:
+            timeout (int): maximum time to wait for the 'ready' state.
+            sleep (int): sleep time between each sample.
+
+        Returns:
+            bool: True if machine-set reached 'ready' state, False otherwise.
+        """
+        try:
+            for ready_replicas in TimeoutSampler(
+                wait_timeout=timeout, sleep=sleep, func=lambda: self.ready_replicas
+            ):
+                if ready_replicas and ready_replicas == self.desired_replicas:
+                    return True
+        except TimeoutExpiredError:
+            LOGGER.error(
+                f"Ready replicas: {self.ready_replicas}, desired replicas: {self.desired_replicas}"
+            )
+            return False
+
+    def scale(self, replicas, wait_timeout=300, sleep=1):
+        """
+        Scale down/up a machine-set.
+
+        Args:
+            replicas (int): num of replicas to scale to.
+            wait_timeout (int): maximum time to wait for scaling the machine-set.
+            sleep (int): sleep time between each sample of the machine-set state.
+
+        Returns:
+            bool: True if scaling the machine set was successful, False otherwise.
+        """
+        LOGGER.info(
+            f"Scale machine-set from {self.desired_replicas} replicas to {replicas} replicas"
+        )
+
+        return self.update(
+            patches={"spec": {"replicas": replicas}},
+            wait_timeout=wait_timeout,
+            sleep=sleep,
+        )
+
+    def update(self, patches, wait_timeout=TIMEOUT, sleep=1):
+        """
+        Update machine-set.
+
+        Args:
+            patches (dict): information to update the machine-set.
+            wait_timeout (int): maximum time to wait for machine-set to be ready after updating.
+            sleep (int): sleep time between each sample of the machine-set state.
+
+        Returns:
+            bool: True if updating the machine-set was successful, False otherwise.
+        """
+        LOGGER.info(f"Update machine-set {self.name} to {patches}")
+        ResourceEditor(patches={self: patches}).update()
+        return self.wait(timeout=wait_timeout, sleep=sleep)
+
+    def to_dict(self):
+        res = super().to_dict()
+        if self.yaml_file:
+            return res
+
+        _spec, _metadata, _labels, _template = (
+            "spec",
+            "metadata",
+            "labels",
+            "template",
+        )
+        (
+            _cluster_api_cluster,
+            _cluster_api_machine_role,
+            _cluster_api_machine_type,
+            _cluster_api_machineset,
+        ) = (
+            "cluster-api-cluster",
+            "cluster-api-machine-role",
+            "cluster-api-machine-type",
+            "cluster-api-machineset",
+        )
+
+        res[_metadata].setdefault(_labels, {}).update(
```

Comment:
Discard my previous comments please, make is simple
no need to use setdefault(), just build you dict as needed.
we can assume that anyone that cal to_dict() create new resource.

Reply:
ack